### PR TITLE
cli / codespaces update

### LINKED ISSUE #208: add env vars during flt create
- the factory AI app needs additional env vars for setup

It's easy enough to modify the setup scripts to add

I'm wondering if we shouldn't allow passing in multipe env vars on the command line? That seems easier and more reusable as you wouldn't have to have a separate CLI branch.

### LINKED ISSUE #235: group flt azure commands
- flt az
  - groups
  - login
  - logout

### LINKED ISSUE #236: flt create - use env var
- use the AKDC_SSL and AKDC_GITOPS as defaults
- simplifies and reduces typos

### LINKED ISSUE #237: only run apt-get upgrade on prebuilds
- rebuilding a CS image takes forever when az cli updates

### LINKED ISSUE #239: flt create fails
- vm/setup/bootstrap is empty
- cause kubectl apply -f bootstrap to fail

### PR DESCRIPTION
**Pull request by bartr**
## Purpose of PR

- GitOps with private repos
- group `flt az` commands
- `flt create` use env vars for defaults
- codespaces - only run `apt-get upgrade` on pre-build
- add AKDC_* env vars during fleet create

## Type of PR

- [ ] Documentation changes
- [x] Code changes
- [x] Test changes
- [ ] CI-CD changes
- [x] Codespaces changes

## Issues Closed or Referenced

- Closes #208 
- Closes #223 
- Closes #235 
- Closes #236 
- Closes #237 
- Closes #239

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -9,16 +9,9 @@ ARG USERNAME=vscode
 # log docker build start
 RUN echo "$(date +'%Y-%m-%d %H:%M:%S')    docker build start" >> "/home/${USERNAME}/status"
 
-# install fluent-bit
```
Comment (bartr) [inline .devcontainer/Dockerfile:12]: installed in DinD image base

File: bin/.flt/commands/az/groups [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
```
Comment (bartr) [inline bin/.flt/commands/az/groups:1]: this was moved unchanged

File: src/kic/boa/boa.go [hunk 3]
```diff
@@ -433,13 +433,25 @@ func addFltCommand(use string, short string, long string, command string) *cobra
 		Long:  long,
 
 		RunE: func(cmd *cobra.Command, args []string) error {
-			script := filepath.Join(".", "cli", "vm", "scripts", command)
```
Comment (bartr) [inline src/kic/boa/boa.go:436]: the gisv repo has the cli in the gitops repo for simplicity

File: src/kic/boa/boa.go [hunk 4]
```diff
@@ -433,13 +433,25 @@ func addFltCommand(use string, short string, long string, command string) *cobra
...
+			}
+
+			script = filepath.Join(".", script, "vm", "scripts", command)
 
 			// add the paramaters
 			if len(args) > 0 {
 				script += " " + strings.Join(args, " ")
 			}
 
+			// check-setup can run before the cli is fully setup
```
Comment (bartr) [inline src/kic/boa/boa.go:450]: this allows flt check setup to run before the CLI repo is cloned - better for debugging setup issues

File: src/kic/cmd/fleet/create.go [hunk 5]
```diff
@@ -51,13 +51,24 @@ var (
 
 // add kic fleet create specific flags
 func init() {
```
Comment (bartr) [inline src/kic/cmd/fleet/create.go:53]: read values from env vars and use as defaults - command line args override

this is temporary until we can add viper

File: vm/setup/akdc-dns.sh [hunk 6]
```diff
@@ -1,11 +1,11 @@
 #!/bin/bash
```
Comment (bartr) [inline vm/setup/akdc-dns.sh:2]: simplified the output file since all the scripts run as akdc user

File: vm/setup/flux-setup.sh [hunk 7]
```diff
@@ -59,12 +59,17 @@ until [ $status_code == 0 ]; do
 done
 
 echo "adding flux sources"
-echo "$(date +'%Y-%m-%d %H:%M:%S')  adding flux sources" >> "/home/${AKDC_ME}/status"
```
Comment (bartr) [inline vm/setup/flux-setup.sh:62]: this is what supports private GitOps repos

File: vm/setup/k8s-setup.sh [hunk 8]
```diff
@@ -34,15 +34,15 @@ sleep 5
...
 if [ "$AKDC_DAPR" = "true" ]
 then
-  echo "$(date +'%Y-%m-%d %H:%M:%S')  installing dapr" >> "/home/${AKDC_ME}/status"
+  echo "$(date +'%Y-%m-%d %H:%M:%S')  installing dapr" >> "$HOME/status"
   wget -q https://raw.githubusercontent.com/dapr/cli/master/install/install.sh -O - | sudo /bin/bash
   sudo dapr init -k --enable-mtls=false --wait
 fi
 
-echo "$(date +'%Y-%m-%d %H:%M:%S')  k3d-setup complete" >> "/home/${AKDC_ME}/status"
+echo "$(date +'%Y-%m-%d %H:%M:%S')  k3d-setup complete" >> "$HOME/status"
```
Comment (bartr) [inline vm/setup/k8s-setup.sh:48]: ns-nuke sometimes doesn't work and sometimes has bad side-effects - it's way safer to run kic cluster recreate

File: bin/.flt/fleet-vm.templ [hunk 9]
```diff
@@ -1,20 +1,31 @@
...
-export AKDC_CLUSTER="{{cluster}}"
-export AKDC_ARC_ENABLED="{{arcEnabled}}"
-export AKDC_RESOURCE_GROUP="{{group}}"
-export AKDC_DO="{{do}}"
-export AKDC_ZONE="{{zone}}"
-export AKDC_DNS_RG="{{dnsRG}}"
-
-###################################
+export HOME="/home/$AKDC_ME"
+
```
Comment (bartr) [inline bin/.flt/fleet-vm.templ:6]: - simplified and extended env vars
- all customizations are in $HOME/akdc.bashrc
- /etc/bash.bashrc sources akdc.bashrc
- {{environment}} gets replaced with export AKDC_*, so it's easy to extend
- completions get sourced in akdc-install.sh so you don't get errors before everything is installed

File: src/kic/cmd/fleet/create.go [hunk 10]
```diff
@@ -446,17 +457,41 @@ func createVMSetupScript() {
...
+			split := strings.SplitN(val, "=", 2)
+			if len(split) == 2 {
+				val = strings.ReplaceAll(split[1], "\n", "")
+				val = strings.ReplaceAll(val, "\"", "\\\"")
+				line := fmt.Sprintf("  echo 'export %s=\"", split[0])
+				line += val
+				line += "\"'\n"
+
+				if split[0] == "AKDC_FOO" {
+					fmt.Println(val)
```
Comment (aflinchb) [inline src/kic/cmd/fleet/create.go:486]: Is this "AKDC_FOO" left over from testing - I see the testing todo above, but wasn't sure if this was included.

Comment (bartr) [reply src/kic/cmd/fleet/create.go:486]: good catch ... oops

**Review by aflinchb — APPROVED** [review]

**Pull request merged**